### ISSUE TITLE
[Linux] Some bugs

### ISSUE THREAD

**Issue by Enderteck** [TeamHypersomnia/Hypersomnia#282]
I already wrote all of them in this Steam discussion:

https://steamcommunity.com/app/2660970/discussions/0/4291440317148295160/

Pasted it here if you're lazy:

Setup

Fedora Linux 39

Latest Nvidia drivers

I've only gone through a portion of the tutorial but I found Some issues with the game.

- When launching the game for the first time, the window is streched betwin two screens.

- Fullscreen is forced to be on my laptop monitor and not on my main screen. I can't change
that for some reason.

- I can't bind the number keys on my keyboard as in the French keyboard, if you don't press
shift you don't get numbers you get [] (1-9 and 0) in this order. When trying to
set any of these keybinds, it says invalid. I can't really play withou thoses keys as scrolling
doesn't change your weapon. There are also some action that require pressing them such
changeing weapon hand and selecting another weapon than the last used weapon.

**Comment by geneotech** [comment]
Hey! Thanks for reporting, indeed I didn't have notifications for Steam forum. Right now it'll be impractical for me to fix the window/fs issues as I don't have a fedora laptop with two screens, just a PC with one monitor.

I'll take a look at the french layout, I'll see if I can add additional keycodes for the characters under numbers in French layout. Could you tell me how does using/binding numbers work in other games in that case? Shift is another keybind so you can't use it to bind actual numbers. Does pressing 1,2,3,4 in say CS settings just bind them to &,é,#,{,[, etc?

**Comment by Enderteck** [comment]
Yes games usually support setting the French number keys symbols.

Some of them d__isplay the English layout (the numbers) while still associated with the symbols but it works.

**Comment by geneotech** [comment]
Hey there! I deployed an update across all platforms (incl. Steam) - Linux build now uses GLFW instead of my crudely written X Widnow system implementation, so this should fix literally all of your issues, the weird fullscreen behavior but especially the French layout handling :) It should now detect number keys as numbers. Thanks again for reporting it, it's a bug that might've prevented an entire demographic from playing the game.

Linux user tip - if like me you've mapped Caps lock to ESC via an X keysym, the game might not recognize it, but there's an option to "remap CAPS LOCK to ESC" in Hypersomnia (Settings->Advanced).

Let me know how the game works for you now!

**Comment by Enderteck** [comment]
Ok I'll try ! Thx

**Comment by Enderteck** [comment]
Yeah definitly works now, great. Thx for making awesome Open Source games !

**Comment by geneotech** [comment]
Glad you like it! :heart:  See you on the game server :)

**Comment by Enderteck** [comment]
Oh I realized the game doesn't run natively under wayland and setting GLFW_PLATFORM_WAYLAND %command% on Steam does nothing. Does the GLFW version used not work with wayland ? I f it does is there a way to make it run natively ? Thx for any answers.

**Comment by geneotech** [comment]
Yes, the default GLFW installation does not come with Wayland support. I'd have to build GLFW manually to enable support. I'll later try instaling libglfw3-wayland instead of libglfw3 on the Ubuntu builder.

**Comment by geneotech** [comment]
It appears I'll have to build it manually to support both X11 and Wayland, you can temporarily use Hypersomnia-Steam.AppImage from this run's artifact:

https://github.com/TeamHypersomnia/Hypersomnia/actions/runs/8383186513

**Comment by Enderteck** [comment]
Ok thanks

**Comment by geneotech** [comment]
Alright. I managed to build GLFW with *both* X11 and Wayland support. This build launches on my X11 setup - please see if this build correctly runs Wayland natively:
https://github.com/TeamHypersomnia/Hypersomnia/actions/runs/8383487739

If it does, I'll deploy it to production.
Thank you!

**Comment by Enderteck** [comment]
Works great for me on Fedora 39, both appimages ! Thanks

**Comment by geneotech** [comment]
Awesome! Thank you so much for your assistance :) :heart:
